src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/QueryExecutor.java

```diff
@@ -42,14 +42,25 @@ public QueryExecutor(OperationsConfig operationsConfig, StargateRequestInfo star
    * @param pagingState - In case of pagination, the paging state needs to be passed to fetch
    *     subsequent pages
    * @param pageSize - page size
+   * @param vectorSearch - whether the query uses vector search -- use ANN OF
    * @return AsyncResultSet
    */
   public Uni<AsyncResultSet> executeRead(
-      SimpleStatement simpleStatement, Optional<String> pagingState, int pageSize) {
-    simpleStatement =
-        simpleStatement
-            .setPageSize(pageSize)
-            .setConsistencyLevel(operationsConfig.queriesConfig().consistency().reads());
+      SimpleStatement simpleStatement,
```

Comment:
I wonder if it'd be cleaner to have 2 methods -- executeRead() and executeVectorSearch() instead of one more parameter? Mostly it would be more obvious from call since I think it is easy to get two mixed.
(note: I am not sure this is better way, just suggesting as consideration)

@maheshrajamani WDYT?

Reply:
Changed the boolean flag to the additional method. @tatu-at-datastax Could you help me do the last review? Thanks!

Reply:
Ok @maheshrajamani did the review -- looks good to me too, thank you for doing this @Hazel-Datastax !